### ISSUE TITLE
test_lambda_duoble fails when WITH_LLVM=yes

### ISSUE THREAD

**Issue by erikagnvall** [symengine/symengine#1224]
The `test_lambda_double` test seems to fail when SymEngine is built with LLVM. Steps I took:

* Checkout SymEngine master (ff148c1f0d28e374862b27a89e4ad61e5d305fe2 at the time of writing)
* Build and install LLVM (3.9.0) from source
* Compile SymEngine (with `WITH_LLVM=yes`, *this is the important part*)
* Run the tests with `ctest`

Result (irrelevant output removed):
```
Test project /home/erik/src/symengine
      Start  1: test_rcp
 1/35 Test  #1: test_rcp .........................   Passed    0.00 sec
[...]
      Start 31: test_lambda_double
31/35 Test #31: test_lambda_double ...............***Failed    0.04 sec
[...]

97% tests passed, 1 tests failed out of 35

Total Test time (real) =   0.54 sec

The following tests FAILED:
         31 - test_lambda_double (Failed)
Errors while running CTest
```

Specifically running the `test_lambda_double` executable outputs the following:
```
Intrinsic name not mangled correctly for type arguments! Should be: llvm.powi.f64
double (double, i32)* @llvm.powi.f64.i32
Intrinsic name not mangled correctly for type arguments! Should be: llvm.pow.f64
double (double, double)* @llvm.pow.f64.f64
LLVM ERROR: Broken function found, compilation aborted!
```

Unfortunately, I'm not familiar enough with LLVM to know how to fix it.

**Comment by ShikharJ** [comment]
@Meldanya I think these errors are because of the LLVM version being used. We currently test 3.8.0 if I'm not mistaken (I couldn't reproduce the errors on my machine). However, I'm not much confident to tell what can be done to fix this, or whether a probable fix will be backwards compatible with 3.8.0.

**Comment by Meldanya** [comment]
@ShikharJ As far as I could tell from the Travis logs, SymEngine isn't tested with the LLVM flag enabled? I'll try testing against LLVM 3.8.0 and let you know if I'm able to reproduce the problem.

**Comment by ShikharJ** [comment]
@Meldanya Please take a look [here](https://travis-ci.org/symengine/symengine/jobs/212039907/config). Build 15 contains the `WITH_LLVM=yes` clause in the `env` line. For the version of LLVM used you can refer [here](https://github.com/symengine/symengine/blob/master/bin/install_travis.sh#L96-L97). 
Also, keep us posted if you face any more issues.

**Comment by Meldanya** [comment]
@ShikharJ Ah, sorry, I totally missed that!

However, I can't get it to work with LLVM 3.8.0 either. I get the same error. I didn't install with Conda though, I build LLVM from the source tar file [here](http://releases.llvm.org/3.8.0/llvm-3.8.0.src.tar.xz). Can't see how that should make any difference though?

Do you have any ideas for further troubleshooting?

**Comment by ShikharJ** [comment]
Unfortunately, I don't. Maybe @isuruf can help.

**Comment by isuruf** [comment]
@Meldanya, I can't reproduce your issue. Can you give us a bit more info on your system and also  how you compiled LLVM and symengine?

**Comment by Meldanya** [comment]
Sorry for the delay, sure I can provide more details. Let me know any other information you need:

``` sh
$ cat /etc/issue
Ubuntu 16.04.2 LTS \n \l
$ uname -r
4.8.0-36-generic
$ cc --version
cc (Ubuntu 5.4.0-6ubuntu1~16.04.4) 5.4.0 20160609
[...]
$ cmake --version
cmake version 3.5.1
[...]
```

Just to make sure there isn't anything weird with my system, I created a new VM in VirtualBox with Ubuntu 16.04.2 and ran the following script:

```sh
set -x
sudo apt install libgmp-dev cmake make git curl
mkdir -p ~/src && cd ~/src

# Fetch and unpack LLVM
curl http://releases.llvm.org/3.8.0/llvm-3.8.0.src.tar.xz | tar xJ
# Build and install LLVM
cd llvm-3.8.0.src &&
	mkdir -p build &&
	cd build &&
	cmake .. &&
	make -j2 &&
	sudo make install

cd ~/src

# Clone and build SymEngine
git clone https://github.com/symengine/symengine.git &&
	cd symengine &&
	cmake -DWITH_LLVM=yes . &&
	make &&
	ctest
```

I still get the same error.

When you run the tests, you install LLVM with Conda right? Could LLVM have been built with different options in the Conda package? I let CMake pick all the default options when I built it. I'm not very familiar with Conda so I wasn't able to find any manifest for how the `llvmdev` package was built.

**Comment by isuruf** [comment]
Thanks for the detailed instructions. I can reproduce the error. I'll look into this more.

Btw, script for how conda builds it is here, https://github.com/conda-forge/llvmdev-feedstock/blob/master/recipe/build.sh

**Comment by isuruf** [comment]
When I build LLVM in Release mode, it works fine. Not sure why it fails in debug mode though

**Comment by Meldanya** [comment]
@isuruf I just tried building LLVM in Release mode and that works for me as well.

Thanks for the help to find a workaround. I'll let you decide if you want to keep the issue open, from my point of view it could be closed now.

**Comment by lkeegan** [comment]
fixed in #1751
